flink/v1.15/flink/src/main/java/com/netease/arctic/flink/read/hybrid/assigner/StaticSplitAssigner.java

```diff
@@ -0,0 +1,133 @@
+package com.netease.arctic.flink.read.hybrid.assigner;
+
+import com.netease.arctic.flink.read.hybrid.split.ArcticSplit;
+import com.netease.arctic.flink.read.hybrid.split.ArcticSplitState;
+import org.apache.flink.api.connector.source.SplitEnumeratorContext;
+import org.apache.flink.util.FlinkRuntimeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.PriorityBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+
+public class StaticSplitAssigner implements SplitAssigner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(StaticSplitAssigner.class);
+
+  private static final long POLL_TIMEOUT = 200;
+  private final SplitEnumeratorContext<ArcticSplit> enumeratorContext;
+
+  private int totalParallelism;
+  private int totalSplitNum;
+
+  private final PriorityBlockingQueue<ArcticSplit> splitQueue;
+
+  private CompletableFuture<Void> availableFuture;
+
+  public StaticSplitAssigner(
+      SplitEnumeratorContext<ArcticSplit> enumeratorContext) {
+    this.enumeratorContext = enumeratorContext;
+    this.totalParallelism = enumeratorContext.currentParallelism();
+    this.splitQueue = new PriorityBlockingQueue<>();
+  }
+
+  public StaticSplitAssigner(
+      SplitEnumeratorContext<ArcticSplit> enumeratorContext,
+      Collection<ArcticSplitState> splitStates) {
+    this.enumeratorContext = enumeratorContext;
+    this.splitQueue = new PriorityBlockingQueue<>();
+    splitStates.forEach(state -> onDiscoveredSplits(Collections.singleton(state.toSourceSplit())));
+  }
+
+  @Override
+  public Split getNext() {
+    return getNextSplit().map(Split::of).orElseGet(isEmpty() ? Split::unavailable : Split::subtaskUnavailable);
```

Comment:
```suggestion
    return getNextSplit().map(Split::of).orElseGet(Split::unavailable);
```
I don't think StaticSplitAssigner will return subtaskunvailable state, this state should only be returned for ShuffleSplitAssigner.